File: src/types/document.ts
// ===== 文档类型定义 - 与 Rust 后端对齐 =====

export interface PreviewContent {
  path: string;
  fileType: string;
  content: string;
  pageCount?: number;
  sheetNames?: string[];
  metadata?: DocumentMetadata;
}

export interface DocumentMetadata {
  title?: string;
  author?: string;
  created?: string;
  modified?: string;
  wordCount?: number;
}

export interface VersionInfo {
  versionId: string;
  path: string;
  timestamp: string;
  operation: string;
  description: string;
  size: number;
  sessionId?: string;
}

File: src-tauri/src/models/document.rs
use serde::{Deserialize, Serialize};

/// 文档预览内容
#[derive(Serialize, Deserialize, Debug, Clone)]
#[serde(rename_all = "camelCase")]
pub struct PreviewContent {
    pub path: String,
    /// "docx" | "xlsx" | "pptx" | "pdf" | "md"
    pub file_type: String,
    /// 预览文本内容
    pub content: String,
    /// 页数（适用于 PDF/DOCX）
    pub page_count: Option<u32>,
    /// 工作表名称（适用于 XLSX）
    pub sheet_names: Option<Vec<String>>,
    pub metadata: Option<DocumentMetadata>,
}

/// 文档元数据
#[derive(Serialize, Deserialize, Debug, Clone)]
#[serde(rename_all = "camelCase")]
pub struct DocumentMetadata {
    pub title: Option<String>,
    pub author: Option<String>,
    pub created: Option<String>,
    pub modified: Option<String>,
    pub word_count: Option<u32>,
}
